contracts/marketplace/nft_marketplace.ral

```diff
@@ -61,10 +60,17 @@ Contract NFTMarketPlace(
         let price = nftListing.getPrice()
         let commissionRateOfListing = nftListing.getCommissionRate()
         let listingFeeOfListing = nftListing.getListingFee()
+        let royaltyAmount = nftListing.royaltyAmount()
+        let priceAfterFee = checkListingPriceAfterFee(price, commissionRateOfListing, listingFeeOfListing, royaltyAmount)
 
-        let priceAfterFee = checkListingPriceAfterFee(price, commissionRateOfListing, listingFeeOfListing)
         nftListing.buy{buyer -> ALPH: priceAfterFee}(buyer)
 
+        if (royaltyAmount > 0) {
+            let collectionWithRoyalty = INFTCollectionWithRoyalty(NFT(tokenId).getCollectionId())
+            //Report: [API Error] - Execution error when estimating gas for tx script or contract: InvalidExternalMethodArgLength
```

Comment:
Does this error still exist?

Reply:
It appears once, but I can't reproduce it anymore, maybe it has to do with the mismatch of the artifacts, I am not 100% sure. I will remove the comment for now.